job.py

```diff
@@ -33,6 +33,17 @@ def is_plotting_cmdline(cmdline):
         and 'create' == cmdline[3]
     )
 
+# This is a (temporary?) fix for https://github.com/ericaltendorf/plotman/issues/41
+def cmdline_argfix(cmdline):
+    known_keys = 'krbut2dne'
+    for i in cmdline:
+        if i[0]=='-' and i[1] in known_keys and len(i)>2:
+            key = i[0:2]
+            val = i[2:]
+            cmdline[cmdline.index(i)] = key
+            cmdline.insert(cmdline.index(key)+1, val)
```

Comment:
I would tend to avoid the mutation of the `cmdline` argument here.  This could instead either create a new list to return or yield out the elements as a generator.  Mutating arguments can be confusing and surprising to callers.  Also, this loop is mutating the list it is iterating over.  I think that the next iteration will actually be checking the newly-inserted element.  This is probably not a problem in most cases, but doesn't seem likely to be good or wanted.

If the mutation were to be kept, the loop could be `for index, element in enumerate(cmdline):` to avoid the multiple `cmdline.index()` calls that are searching through the list (and hopefully finding the correct element and not some other that matches).

Reply:
True, it's a bit messy. Will implement it differently when I redo this on the development branch.